File: clients/ruby/api_particulier/examples/basic.rb
#!/usr/bin/env ruby
# frozen_string_literal: true
#
# Basic happy path against staging.
#   TOKEN=$(curl -s https://raw.githubusercontent.com/datagouv/apistration/develop/mocks/tokens/default)
#   API_PARTICULIER_TOKEN=$TOKEN bundle exec ruby examples/basic.rb

$LOAD_PATH.unshift File.expand_path('../lib', __dir__)
require 'api_particulier'

client = ApiParticulier::Client.new(
  environment: :staging,
  default_params: { recipient: '13002526500013' }
)

response = client.ants.extrait_immatriculation_vehicule(immatriculation: 'AA-123-BB')

puts "status:    #{response.http_status}"
puts "remaining: #{response.rate_limit&.remaining}"
puts "data keys: #{response.data&.keys.inspect}"

File: clients/node/api-particulier/examples/basic.ts
import { Client } from '../src/index.js';

const client = new Client({
  token: process.env.API_PARTICULIER_TOKEN,
  environment: 'staging',
  defaultParams: {
    recipient: '13002526500013',
  },
});

const response = await client.get('/v3/dss/quotient_familial/identite', {
  params: {
    nom: 'Dupont',
    prenoms: ['Jean'],
    date_de_naissance: '1990-01-01',
    code_cog_commune_naissance: '75056',
  },
});

console.log('Status:', response.httpStatus);
console.log('Data:', JSON.stringify(response.data, null, 2));
console.log('Rate limit:', response.rateLimit);
